eva-accession-deprecate/src/main/java/uk/ac/ebi/eva/accession/deprecate/io/ClusteredVariantsToDeprecateReader.java

```diff
@@ -0,0 +1,135 @@
+
+    private static final String REFERENCE_ASSEMBLY_FIELD = "asm";
+
+    private static final String TAXONOMY_FIELD = "tax";
+
+    private static final String CONTIG_FIELD = "contig";
+
+    private static final String START_FIELD = "start";
+
+    private static final String TYPE_FIELD = "type";
+
+    private static final String ACCESSION_KEY = "accession";
+
+    private static final String VERSION_KEY = "version";
+
+    private MongoClient mongoClient;
+
+    private String database;
+
+    private MongoCursor<Document> cursor;
+
+    public ClusteredVariantsToDeprecateReader(MongoClient mongoClient, String database) {
+        this.mongoClient = mongoClient;
+        this.database = database;
+    }
+
+    @Override
+    public void open(ExecutionContext executionContext) throws ItemStreamException {
+        MongoDatabase db = mongoClient.getDatabase(database);
+        MongoCollection<Document> collection = db.getCollection(DBSNP_CLUSTERED_VARIANT_ENTITY_DECLUSTERED);
+        AggregateIterable<Document> declusteredVariants = collection.aggregate(buildAggregation())
+                                                                    .allowDiskUse(true)
+                                                                    .useCursor(true);
+        cursor = declusteredVariants.iterator();
+    }
+
+    private List<Bson> buildAggregation() {
+        Bson lookup = Aggregates.lookup(DBSNP_CLUSTERED_VARIANT_ENTITY, ACCESSION_FIELD, ACCESSION_FIELD, RS_FIELD);
+        Bson match = Aggregates.match(Filters.eq(RS_FIELD, Collections.EMPTY_LIST));
+        List<Bson> aggregation = Arrays.asList(lookup, match);
+        logger.info("Issuing aggregation: {}", aggregation);
+        return aggregation;
+    }
+
+    @Override
+    public DbsnpClusteredVariantEntity read() {
+        return cursor.hasNext() ? getClusteredVariantToDeprecate(cursor.next()) : null;
+    }
+
+    private DbsnpClusteredVariantEntity getClusteredVariantToDeprecate(Document declusteredVariant) {
+        String id = declusteredVariant.getString(ID_FIELD);
+        String assembly = declusteredVariant.getString(REFERENCE_ASSEMBLY_FIELD);
+        int taxonomy = declusteredVariant.getInteger(TAXONOMY_FIELD);
+        String contig = declusteredVariant.getString(CONTIG_FIELD);
+        long start = declusteredVariant.getLong(START_FIELD);
+        String type = declusteredVariant.getString(TYPE_FIELD);
+        long accession = declusteredVariant.getLong(ACCESSION_KEY);
+        int version = declusteredVariant.getInteger(VERSION_KEY);
+        LocalDateTime createdDate = declusteredVariant.getDate("createdDate")
+                                                      .toInstant().atZone(ZoneId.systemDefault()).toLocalDateTime();
+
+        return new DbsnpClusteredVariantEntity(accession, id, assembly, taxonomy, contig, start,
+                                               VariantType.valueOf(type), false, createdDate, version);
```

Comment:
Validated flag is explicitly set to false here. Even though the validated flag for a variant in the deprecated collection might not be very important, let's just preserve what was in the source for data integrity reasons.

Reply:
in fact, maybe we can use the converter from mongoTemplate, but it uses the old mongo api with DBObjects instead of Documents.
```
        MongoConverter converter = mongoTemplate.getConverter();
        DbsnpClusteredVariantEntity read = converter.read(DbsnpClusteredVariantEntity.class, new BasicDBObject());
```
so either keep this as it is, or do the aggregation with the old api and the automatic converter